core/src/main/java/org/radargun/stages/control/RepeatEndStage.java

```diff
@@ -26,17 +26,19 @@ private void updateState(StateBase state) {
       log.trace("Removing counter " + counterName);
       state.remove(counterName);
       Stack<String> repeatNames = (Stack<String>) state.get(RepeatStage.REPEAT_NAMES);
-      if (repeatNames == null || repeatNames.isEmpty() || repeatNames.peek() != name) {
+      if (repeatNames == null || repeatNames.isEmpty() || !equals(repeatNames.peek(), name)) {
          throw new IllegalStateException("Cannot unwind repeat: " + repeatNames);
       }
       repeatNames.pop();
    }
 
+   private static boolean equals(String s1, String s2) {
```

Comment:
Wait, this is from Objects.equals, why not using it right away when we support 1.7? :-)